samtranslator/model/api/api_generator.py

```diff
@@ -287,9 +287,28 @@ def _add_auth(self):
             self._set_default_authorizer(swagger_editor, authorizers, auth_properties.DefaultAuthorizer)
 
         # Assign the Swagger back to template
+
         self.definition_body = self._openapi_auth_postprocess(swagger_editor.swagger)
 
     def _openapi_auth_postprocess(self, definition_body):
+        """
+        Convert auth components to openapi 3 in definition body if OpenApiVersion flag is specified.
+
+        To maintain backward compatibility with existing swagger documents that contain both swagger
+        and openapi defined in the definition body, we return if there is "swagger" defined in the
+        definition body.
+
+        If the definition body contains "openapi" and not "swagger", we enforce that the OpenApiVersion
```

Comment:
Why do we enforce this case?

Reply:
> Why do we enforce this case?

Read Brett's comment above. Additionally, if we are generating/adding to the definitionBody, we make sure the OpenApiVersion flag is specified before we decide to go with OAS3.